app/utils/detectors.py

```diff
@@ -0,0 +1,130 @@
+import re
+from typing import List, Optional
+
+import pandas as pd
+import pyarrow as pa
+from fastapi import HTTPException
+
+ID_KEY = "id"
+VALUE_KEY = "value"
+DATE_KEYS = ["date", "day", "time", "timestamp", "datetime", "dttm"]
+
+
+def _norm(name: str) -> List[str]:
+    name = str(name).lower().strip()
+    return re.split(r"[^a-z0-9]+", name)
+
+
+def _find_by_name(
+    table: pa.Table,
+    keys: List[str],
+    *,
+    exclude: Optional[int] = None,
+) -> List[int]:
+    cols: List[int] = []
+    keys = [k.lower() for k in keys]
+
+    for idx, name in enumerate(table.column_names):
+        if exclude is not None and idx == exclude:
+            continue
+
+        tokens = _norm(name)
+
+        for tok in tokens:
+            for key in keys:
+                if tok == key or tok.startswith(key) or tok.endswith(key) or key in tok:
+                    cols.append(idx)
+                    break
+            else:
+                continue
+            break
+
+    return cols
+
+
+def _resolve(cols: List[int], table: pa.Table, label: str) -> Optional[int]:
+    if len(cols) > 1:
```

Comment:
discussion: i'm not sure how useful it is to raise in this situation. maybe return the first element with a warning?